ecommerce/models.py

```diff
@@ -0,0 +1,277 @@
+"""Models for ecommerce"""
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+
+from mitxpro.models import AuditableModel, AuditModel, TimestampedModel
+
+
+class Product(TimestampedModel):
+    """
+    Representation of a purchasable product.
+    """
+
+    content_type = models.ForeignKey(
+        ContentType,
+        on_delete=models.PROTECT,
+        null=True,
+        help_text="content_object is a link to either a Course, CourseRun, or a Program",
+    )
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey("content_type", "object_id")
+
+    class Meta:
+        unique_together = ("content_type", "object_id")
+
+    def __str__(self):
+        """Description of a product"""
+        return f"Product for {self.content_object}"
+
+
+class ProductVersion(TimestampedModel):
+    """
+    An append-only table for Product, keeping track of the parts of a product which are editable.
+    """
+
+    product = models.ForeignKey(
+        Product, on_delete=models.PROTECT, related_name="productversions"
+    )
+    price = models.DecimalField(decimal_places=2, max_digits=20)
+    description = models.TextField()
+
+    class Meta:
+        indexes = [models.Index(fields=["created_on"])]
+
+    def __str__(self):
+        """Description of a ProductVersion"""
+        return f"ProductVersion for {self.description}, ${self.price}"
+
+
+class Basket(TimestampedModel):
+    """
+    Represents a User's basket
+    """
+
+    user = models.OneToOneField(settings.AUTH_USER_MODEL, on_delete=models.PROTECT)
+
+    def __str__(self):
+        """Description of Basket"""
+        return f"Basket for {self.user}"
+
+
+class BasketItem(TimestampedModel):
+    """
+    Represents a product in a user's basket
+    """
+
+    product = models.ForeignKey(Product, on_delete=models.PROTECT)
+    basket = models.ForeignKey(
+        Basket, on_delete=models.PROTECT, related_name="basketitems"
+    )
+    quantity = models.PositiveIntegerField()
+
+    def __str__(self):
+        """Description of BasketItem"""
+        return f"BasketItem for {self.quantity} {self.product}(s)"
+
+
+class Order(TimestampedModel, AuditableModel):
+    """
+    An order containing information for a purchase.
+    """
+
+    FULFILLED = "fulfilled"
+    FAILED = "failed"
+    CREATED = "created"
+    REFUNDED = "refunded"
+
+    STATUSES = [CREATED, FULFILLED, FAILED, REFUNDED]
+    FULFILLED_STATUSES = [FULFILLED]
+
+    CYBERSOURCE = "cybersource"
+    PURCHASE_ORDER = "purchase-order"
```

Comment:
Is there supposed to be an order type field that would be set to one of these two values? Not seeing any usage for these